__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1508 @@
+    Object.defineProperty(vscode.window, "showInformationMessage", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showWarningMessage", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.workspace, "openTextDocument", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showTextDocument", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.commands, "executeCommand", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals, "LOG", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals.LOG, "debug", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals.LOG, "error", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "Download", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.Download, "dataSet", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "Delete", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.Delete, "dataSet", { value: jest.fn(), configurable: true });
+    Object.defineProperty(fs, "unlinkSync", { value: jest.fn(), configurable: true });
+    Object.defineProperty(fs, "existsSync", { value: jest.fn(), configurable: true });
+    Object.defineProperty(sharedUtils, "concatChildNodes", { value: jest.fn(), configurable: true });
+    Object.defineProperty(Profiles, "getInstance", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "List", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.List, "dataSet", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode, "ProgressLocation", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "createWebviewPanel", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.env, "clipboard", { value: clipboard, configurable: true });
+}
+
+// Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
+const mocked = <T extends (...args: any[]) => any>(fn: T): jest.Mock<ReturnType<T>> => fn as any;
+
+describe("Dataset Actions Unit Tests - Function createMember", () => {
+    let blockMocks;
+
+    function createBlockMocks() {
+        const session = generateISession();
+        const imperativeProfile = generateIProfile();
+        const zosmfSession = generateBasicZosmfSession(imperativeProfile);
+        const treeView = generateTreeView();
+        const datasetSessionNode = generateDatasetSessionNode(session, imperativeProfile);
+        const mvsApi = generateMvsApi(imperativeProfile);
+        bindMvsApi(mvsApi);
+
+        return {
```

Comment:
If you create an object with these properties, and just return the object, it would reduce the length of this function by about half. ie:
````
function createBlockMocks() {
    const newMocks = {
        session: generateISession(),
        imperativeProfile: generateIProfile(),
        zosmfSession: generateBasicZosmfSession(imperativeProfile),
        treeView: generateTreeView(),
        datasetSessionNode: generateDatasetSessionNode(session, imperativeProfile),
        mvsApi: generateMvsApi(imperativeProfile),
    }
    bindMvsApi(newMocks.mvsApi);
    return newMocks;
}
````
And then in each it() block, instead of using the variable from the describe() block, you would redeclare blockMocks:
`const blockMocks = createBlockMocks();`

Reply:
It makes sense. On the other hand `datasetSessionNode` requires session to be already defined. And `testDatasetTree` requires `datasetSessionNode ` to be defined. So these have to come before the newMocks object definition. So I would suggest just moving `testDatasetTree` above, like the rest and then return an object composed only of names.